src/SIL.XForge/Services/UserService.cs

```diff
@@ -151,6 +155,23 @@ public async Task LinkParatextAccountAsync(string primaryAuthId, string paratext
         await using IConnection conn = await _realtimeService.ConnectAsync(primaryUserId);
         IDocument<User> userDoc = await conn.FetchAsync<User>(primaryUserId);
         await userDoc.SubmitJson0OpAsync(op => op.Set(u => u.ParatextId, GetIdpIdFromAuthId(ptId)));
+        IDocument<User> secondaryUserDoc = await conn.FetchAsync<User>(secondarySFUserId);
+        if (secondaryUserDoc.Data != null)
+        {
+            this._logger.LogInformation(
+                $"UserService.LinkParatextAccountAsync() will remove the paratextId and tokens from SF user id {secondarySFUserId}. Perhaps a race condition occurred, resulting in a situation like SF-1849. This secondary SF user and associated user_secret can probably be deleted."
+            );
+            await secondaryUserDoc.SubmitJson0OpAsync(op => op.Set(u => u.ParatextId, null));
+        }
+
+        if (await _userSecrets.GetAsync(secondarySFUserId) != null)
+        {
+            await _userSecrets.UpdateAsync(
+                secondarySFUserId,
+                update => update.Set(us => us.ParatextTokens, null),
+                true
```

Comment:
You have retrieved the object in line 167 to check its existence, but here you are upserting in case it does not exist? I would have thought that if the user secret record did not exist (i.e. was somehow deleted by another action on the database at the same time), you would not want to insert the null value?

Reply:
Thank you! By leaving upset set to true, I was creating my own need to first check the existence. Leaving upsert as false is much better.